inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -1187,34 +1189,86 @@ public Predictions computePredictions(User aUser, Project aProject,
         int predictionCount = 0;
 
         Type tokenType = getType(aCas, Token.class);
+        Type sentenceType = getType(aCas, Sentence.class);
 
         List<AnnotationSuggestion> result = new ArrayList<>();
         int id = 0;
+
         for (AnnotationFS annotationFS : CasUtil.select(aCas, predictedType)) {
             if (!annotationFS.getBooleanValue(predictionFeature)) {
                 continue;
             }
 
-            List<AnnotationFS> tokens = CasUtil.selectCovered(tokenType, annotationFS);
-            if (tokens.isEmpty()) {
-                // This can happen if a recommender uses different token boundaries (e.g. if a 
-                // remote service performs its own tokenization). We might be smart here by looking
-                // for overlapping tokens instead of contained tokens.
-                continue;
+            int begin;
+            int end;
+            switch (aRecommender.getLayer().getAnchoringMode()) {
+            case CHARACTERS: {
+                int[] offsets = { annotationFS.getBegin(), annotationFS.getEnd() };
+                TrimUtils.trim(annotationFS.getCAS().getDocumentText(), offsets);
+                begin = offsets[0];
+                end = offsets[1];
+                break;
+            }
+            case SINGLE_TOKEN: {
+                List<AnnotationFS> tokens = selectCovered(tokenType, annotationFS);
+                if (tokens.isEmpty()) {
+                    // This can happen if a recommender uses different token boundaries (e.g. if a
+                    // remote service performs its own tokenization). We might be smart here by
+                    // looking for overlapping tokens instead of contained tokens.
+                    continue;
+                }
+                
+                AnnotationFS firstToken = tokens.get(0);
+                AnnotationFS lastToken = tokens.get(tokens.size() - 1);
+                
+                if (!firstToken.equals(lastToken)) {
+                    // We only want to accept single-token suggestions
```

Comment:
I would maybe add some logging here.

Reply:
We could add debug logging, but I wouldn't add info logging.

Reply:
That would be good, I agree in that info would be too much.